kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -271,11 +303,14 @@ public void initialize() {
               .getProvidedValues();
         }
       }
+      count = 0;
     }
 
     @Override
-    public AvroStorageKey map(E entity) {
-      return key.reuseFor(entity, provided, accessor);
+    public Pair<GenericData.Record, Integer> map(E entity) {
```

Comment:
Not completely familiar with it but relevant code is here:
https://github.com/apache/crunch/blob/master/crunch-core/src/main/java/org/apache/crunch/types/avro/Avros.java#L490